composer/core/data_spec.py

```diff
@@ -171,7 +171,7 @@ def __init__(
         num_tokens: Optional[int] = None,
         device_transforms: Optional[Callable[[Batch], Batch]] = None,
         split_batch: Optional[Callable[[Batch, int], Sequence[Batch]]] = None,
-        get_num_samples_in_batch: Optional[Callable[[Batch], int]] = None,
+        get_num_samples_in_batch: Optional[Callable[[Batch], Union[int, float]]] = None,
```

Comment:
You can leverage the [numerical tower](https://peps.python.org/pep-0484/#the-numeric-tower) as an `int` is a valid `float`.
```suggestion
        get_num_samples_in_batch: Optional[Callable[[Batch], float]] = None,
```

Reply:
Is this the recommended path by PEP? I'm personally more in favor of leaving both to be more explicit to user in type hints, but if PEP recommends using numerical tower, maybe it's better to migrate across repo...